rmf_fleet_adapter/src/full_control/main.cpp

```diff
@@ -213,6 +213,22 @@ class FleetDriverRobotCommandHandle
       location.y = p.y();
       location.yaw = p.z();
 
+      // The speed limit is set as the minimum of all the approach lanes' limits
+      std::optional<double> speed_limit;
+      for (const auto& lane_idx : wp.approach_lanes())
+      {
+        const auto& lane = _travel_info.graph->get_lane(lane_idx);
+        const auto& lane_limit = lane.properties().speed_limit();
+        if (lane_limit.has_value())
+        {
+          if (speed_limit.has_value())
+            speed_limit = std::min(speed_limit.value(), lane_limit.value());
+          else
+            speed_limit = lane_limit.value();
+        }
+      }
+      location.approach_speed = speed_limit.has_value() ? speed_limit.value() : 0.0;
```

Comment:
Wondering if we should use `NaN` or a negative number instead of `0.0`? speed = 0.0 does not intuitively convey that the robot should not respect any speed limits

Reply:
I have similar concerns that I added on the review, https://github.com/open-rmf/rmf_internal_msgs/pull/28